units/doctest.h

```diff
@@ -1,25 +1,33 @@
-// ======================================================================
+// DEAR SEAHORN DEV,
```

Comment:
I see that you updated doctest to the latest version. This is good. However, commit message does not say that.

Do not change this file. At all.
All changes should be outside the file. Talk to me if there is something that you think you cannot do that way.

Reply:
Couple of points.
1. Doctest does not believe in versions and releases. They ask to `wget` at head. If it helps, I can add the commit-hash info in the commit message.
2. `doctest.h` does not follow this project's clang format. The current pre-merge check uses `clang-format-diff` and does not respect `.clang-format` files. Thus, there is no easy way to exclude this file. I would suggest force merging over the pre-merge failure since this file will be upgraded rarely. The alternative is what I have done here.

I will block further changes to this PR pending reply.

Reply:
I would be happy if commit message says "updated doctest to current master". Then I know what to expect.

working around the warning is the way to go. It is our setup issue, not doctest issue. Easier if whenever we use other people code we preserve format and use the code without any changes at all.

Reply:
Removed all changes to doctest (only upgraded to HEAD). 
Made include order consistent as per https://seahornteam.slack.com/archives/DPND2KEJU/p1665394990646389